apps/verification/src/verification.service.ts

```diff
@@ -627,16 +648,15 @@ export class VerificationService {
               p_value: parseInt(attribute.value),
               restrictions: [
                 {
-                  cred_def_id: proofRequestpayload.attributes[index].credDefId ? proofRequestpayload.attributes[index].credDefId : undefined,
-                  schema_id: proofRequestpayload.attributes[index].schemaId
+                  cred_def_id: proofRequestpayload?.proofFormats?.indy?.attributes[index].credDefId ? proofRequestpayload?.proofFormats?.indy?.attributes[index].credDefId : undefined,
```

Comment:
in case the `attributes[index].credDefId: undefined`, does the proofrequest allow any credDef or does the credDef only needs to be `undefined`?

Reply:
No it doesn't allow credDef in this case `attributes[index].credDefId: undefined`.